lib/Dialect/TritonGPU/Transforms/Coalesce.cpp

```diff
@@ -48,70 +57,66 @@ static Value getMemAccessPtr(Operation *op) {
+  // The divisibility in `order[0]` is 16
+  // TODO[goostavz]: confirm the legality of it
+  contiguity[order[0]] = tileShape[order[0]];
+  divisibility[order[0]] = 16 * 8 / elemSizeInBytes;
+  return AxisInfo(contiguity, divisibility, constancy);
+}
+
 struct CoalescePass : public TritonGPUCoalesceBase<CoalescePass> {
   void
   setCoalescedEncoding(ModuleAxisInfoAnalysis &axisInfoAnalysis, Operation *op,
                        int numWarps, int threadsPerWarp,
                        llvm::MapVector<Operation *, Attribute> &layoutMap) {
     Value ptr = getMemAccessPtr(op);
-    auto refType = ptr.getType();
-    if (refType.isa<PointerType>())
-      refType = refType.cast<PointerType>().getPointeeType();
-    auto refTensorType = refType.cast<RankedTensorType>();
-
-    // TODO(Keren): integrate it into AxisInfoAnalysis
-    // Get axis info
-    auto queryAxisInfo = [&](const Value &val) -> AxisInfo {
-      auto valType = val.getType();
-      // Tensor pointer
-      // TODO(Chenggang): encoding for tensor pointers is meaningless, remove
-      // these later while merging into the GitHub main
-      if (auto ptrType = valType.dyn_cast<PointerType>()) {
-        auto tensorTy = ptrType.getPointeeType().dyn_cast<RankedTensorType>();
-        assert(tensorTy);
-        auto makeTensorPtr = getMakeTensorPtrOp(val);
-        auto order = makeTensorPtr.getOrder();
-        auto tileShape = triton::gpu::getShapePerCTA(tensorTy);
-        size_t rank = order.size();
-        auto elemSizeInBytes =
-            tensorTy.getElementType().getIntOrFloatBitWidth() / 8;
-        SmallVector<int64_t> contiguity(rank, 1);
-        SmallVector<int64_t> divisibility(rank, 1);
-        SmallVector<int64_t> constancy(rank, 1);
-        // The contiguity in `order[0]` is `tileShape[order[0]]`
-        // The divisibility in `order[0]` is 16
-        // TODO[goostavz]: confirm the legality of it
-        contiguity[order[0]] = tileShape[order[0]];
-        divisibility[order[0]] = 16 * 8 / elemSizeInBytes;
-        return AxisInfo(contiguity, divisibility, constancy);
-      }
-      // Normal cases
-      assert(valType.isa<RankedTensorType>());
-      return *axisInfoAnalysis.getAxisInfo(val);
-    };
+    auto refTensorType = getTensorType(ptr);
 
     // Get the contiguity order of `ptr`
     SmallVector<unsigned> order;
-    if (auto ptrType = ptr.getType().dyn_cast<PointerType>()) {
-      // Tensor pointer
+    LDBGS("op is:");
+    LLVM_DEBUG(op->dump());
+    if (ptr.getType().isa<PointerType>()) {
       auto makeTensorPtr = getMakeTensorPtrOp(ptr);
       std::copy(makeTensorPtr.getOrder().begin(),
                 makeTensorPtr.getOrder().end(), std::back_inserter(order));
     } else {
       // Normal cases
-      order = argSort(queryAxisInfo(ptr).getContiguity());
+      auto contiguity = axisInfoAnalysis.getAxisInfo(ptr)->getContiguity();
+      order = argSort(contiguity);
+      LLVM_DEBUG(DBGS() << "contiguity is: "; for (const auto &O
```

Comment:
clang-format is not doing a good job with this.  Try it as

```
LLVM_DEBUG({
  DBGS() << "blah: ";
  for (...) { ... }
});
```

Same below.

Reply:
@jlebar This caused clang-format to fail in pre-commit. Let me if I should go back to the clang-formated version. CC @ThomasRaoux

Reply:
you need to us `{` to do what @jlebar suggested.